sdk/aqueduct/table_artifact.py

```diff
@@ -156,6 +172,93 @@ def save(self, config: SaveConfig) -> None:
             ],
         )
 
+    def validate_with_expectation(
+        self,
+        expectation_name: str,
+        expectation_args: Optional[Dict[str, Any]] = None,
+        severity: CheckSeverity = CheckSeverity.WARNING,
+    ) -> CheckArtifact:
+        """Creates a check that validates with the table with great_expectations and its set of internal expectations.
+        The expectations supported can be found here:
+        https://great-expectations.readthedocs.io/en/latest/reference/glossary_of_expectations.html
+        The expectations supported are only those that can support Pandas on the great_expectations backend.
+        E.g. Use a expectation to check all column values are unique.
+        ge_check = table.validate_with_expectation("expect_column_values_to_be_unique", {"column": "fixed_acidity"})
+        ge_check.get() // True or False based on expectation passing
+        Args:
+            expectation_name:
+                Name of built-in expectation to run with great_expectations
+            expectation_args:
+                Dictionary of args to pass into the expectation suite for the expectation being run.
+            severity:
+                Optional severity associated with the check created with this expectations
+        Returns:
+            A check artifact that represent the validation result of running the expectation provided on the table.
+        """
+
+        def great_expectations_check_method(table: pd.DataFrame) -> bool:
```

Comment:
Curious why you chose to create a new suite with a single expectation each time, instead of creating a single suite with multiple expectations? Is there an advantage of associating multiple expectations with a single suite? I'm not to familiar with what Great Expectations recommends.

Reply:
Every check will run its own python process so the suite loaded from the context would be different for each check regardless. So we create one within the executor and check the validation we need to. For multiple validations within a check, this was brought up during planning and we need more first class support for certain representations to ensure that and i think its the aim of one of the upcoming projects. The code snippet itself was pieced together from the GE docs on how to load and validate expectations in memory.